src/pages/index.js

```diff
@@ -1,256 +1,231 @@
-    handleCardLike,
-    checkIfLiked
+    handleCardDelete,
+    handleLikeClick
   );
-  return card.generateCard();
-}
-
-// LIKE
-function checkIfLiked(ifLiked, element) {
-  if (ifLiked === true) {
-    element
-      .querySelector(".heart-button")
-      .classList.toggle("heart-button_active");
-  }
+  return card.getView();
 }
 
-// DELETE
-function handleDeleteClick(cardId) {
-  console.log(cardId);
-  deleteConfirmModal.open(cardId);
-}
-
-/////////////////////////////////////////////////  VALIDATOR  ////////////////////////////////////////////////////////////////////////////////////
-
-// PROFILE
-const editFormValidator = new FormValidator(config, profileEditForm);
-
-// CARD
-const cardFormValidator = new FormValidator(config, cardAddForm);
-
-// AVATAR
-const avatarFormValidator = new FormValidator(config, avatarEditForm);
+//////////////////////////////////////////////////////////////  FORM  //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-avatarFormValidator.enableValidation();
-editFormValidator.enableValidation();
-cardFormValidator.enableValidation();
+// EDIT PROFILE
+const profileEditPopup = new PopupWithForm(
+  "#profile-edit-modal",
+  (formData) => {
+    profileEditPopup.renderLoading(true);
+    api
+      .setUserInfo(formData)
+      .then((userData) => {
+        userInfo.setUserInfo(userData);
+        profileEditPopup.close();
+      })
+      .catch((err) => {
+        console.log("Failed to update user info:", err);
+      })
+      .finally(() => {
+        profileEditPopup.renderLoading(false);
+      });
+  }
+);
 
-///////////////////////////////////////////////////////////  SUBMIT  ////////////////////////////////////////////////////////////////////////////
+// ADD CARD
+const addCardPopup = new PopupWithForm("#add-image-modal", (inputValues) => {
+  addCardPopup.renderLoading(true);
+  api
+    .addCard(inputValues)
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardSection.addItem(cardElement);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.log("Failed to add card:", err);
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false);
+    });
+});
 
-// PROFILE
-function handleProfileEditSubmit({ name, description }) {
-  user.setUserInfoOnSubmit(name, description);
-  api.updateProfile({ name, description }).then((message) => {
-    console.log(message);
+// DELETE CARD
+const deleteCardModal = new PopupWithForm("#remove-card-modal");
+function handleCardDelete(card) {
+  deleteCardModal.open();
+  deleteCardModal.setSubmitAction(() => {
+    deleteCardModal.renderLoading(true);
+    api
+      .deleteCard(card.id)
+      .then(() => {
+        card.deleteCard();
+        deleteCardModal.close();
+      })
+      .catch((err) => {
+        console.log("Failed to delete card:", err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors